components/search/HotelSearch/FilterButton.tsx

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+
+const FilterButton: React.FC = () => {
+  const [buttonOffset, setButtonOffset] = useState(80); // Initial offset from the top of the navbar
+  const [isSticky, setIsSticky] = useState(false); // To check if the button should stick
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const scrollY = window.scrollY;
+
+      // Adjust the offset while scrolling
+      const newOffset = Math.max(80 - scrollY * 1, 0);
+      setButtonOffset(newOffset);
+
+      // Update sticky state based on scroll position
+      setIsSticky(scrollY > 150); // Adjust the threshold for "stickiness" as needed
+    };
+
+    window.addEventListener("scroll", handleScroll);
+
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, []);
```

Comment:
I'm not sure if I remember correctly, but I think stickiness handles this (when to be abs, vs move with screen). For the offset, you can add a margin around the sticky element.